Cells.js

```diff
@@ -90,28 +90,33 @@ export const data = {
   }
 };
 
+(function verifyCells(cells) {
+  for (const cellName in cells) {
+    const cell = cells[cellName];
+    if(!cell.hasOwnProperty('conversions')) continue;
+    const conversions = cell.conversions;
+    for (const conversionName in conversions) {
+      const conversion = conversions[conversionName];
+      if(!conversion.self) {
+        conversion.self = cellName;
+      }
+      for (const which of ['self', 'other']) {
+        const convertName = conversion[which];
+        if(!cells[convertName]) {
+          throw new Error(`Cell '${cellName}' references '${convertName}' which doesn't exist.`);
+        }
+      }
+    }
+  }
+})(data);
+
 export function canPhase(self, other) {
   if(self == null || other == null) return false;
   const selfData = data[self.id];
   const otherData = data[other.id];
   return selfData.hasGravity && otherData.hasGravity && selfData.density > otherData.density;
 }
 
-export function convert(self, other) {
```

Comment:
It's not needed anymore.

Reply:
I guess the question is why did you relocate the functionality?

Reply:
Because now it runs once on startup, and it's guaranteed to run for every conversion, so there isn't the possibility of typoing a conversion, and then not testing it.

Reply:
This goes towards the other section as well: I don't think we need the test code in the production code.